libr/parse/p/parse_x86_pseudo.c

```diff
@@ -8,130 +8,159 @@
-		{ "dec",  "#--", {1}},
-		{ "div",  "# /= #", {1, 2}},
-		{ "fabs",  "abs(#)", {1}},
-		{ "fadd",  "# = # + #", {1, 1, 2}},
-		{ "fcomp",  "var = # - #", {1, 2}},
-		{ "fcos",  "# = cos(#)", {1, 1}},
-		{ "fdiv",  "# = # / #", {1, 1, 2}},
-		{ "fiadd",  "# = # / #", {1, 1, 2}},
-		{ "ficom",  "var = # - #", {1, 2}},
-		{ "fidiv",  "# = # / #", {1, 1, 2}},
-		{ "fidiv",  "# = # * #", {1, 1, 2}},
-		{ "fisub",  "# = # - #", {1, 1, 2}},
-		{ "fnul",  "# = # * #", {1, 1, 2}},
-		{ "fnop",  " ", {0}},
-		{ "frndint",  "# = (int) #", {1, 1}},
-		{ "fsin",  "# = sin(#)", {1, 1}},
-		{ "fsqrt",  "# = sqrt(#)", {1, 1}},
-		{ "fsub",  "# = # - #", {1, 1, 2}},
-		{ "fxch",  "#,# = #,#", {1, 2, 2, 1}},
-		{ "idiv",  "# /= #", {1, 2}},
-		{ "imul",  "# = # * #", {1, 2, 3}},
-		{ "in",   "# = io[#]", {1, 2}},
-		{ "inc",  "#++", {1}},
-		{ "ja", "if (((unsigned) var) > 0) goto #", {1}},
-		{ "jb", "if (((unsigned) var) < 0) goto #", {1}},
-		{ "jbe", "if (((unsigned) var) <= 0) goto #", {1}},
-		{ "je", "if (!var) goto #", {1}},
-		{ "jg", "if (var > 0) goto #", {1}},
-		{ "jge", "if (var >= 0) goto #", {1}},
-		{ "jle", "if (var <= 0) goto #", {1}},
-		{ "jmp",  "goto #", {1}},
-		{ "jne", "if (var) goto #", {1}},
-		{ "lea",  "# = #", {1, 2}},
-		{ "mov",  "# = #", {1, 2}},
-		{ "movq",  "# = #", {1, 2}},
-		{ "movaps",  "# = #", {1, 2}},
-		{ "movups",  "# = #", {1, 2}},
-		{ "movsd",  "# = #", {1, 2}},
-		{ "movsx","# = #", {1, 2}},
-		{ "movsxd","# = #", {1, 2}},
-		{ "movzx", "# = #", {1, 2}},
-		{ "movntdq", "# = #", {1, 2}},
-		{ "movnti", "# = #", {1, 2}},
-		{ "movntpd", "# = #", {1, 2}},
-		{ "pcmpeqb", "# == #", {1, 2}},
-
-		{ "movdqu", "# = #", {1, 2}},
-		{ "movdqa", "# = #", {1, 2}},
-		{ "pextrb", "# = (byte) # [#]", {1, 2, 3}},
-		{ "palignr", "# = # align #", {1, 2, 3}},
-		{ "pxor", "# ^= #", {1, 2}},
-		{ "xorps", "# ^= #", {1, 2}},
-		{ "mul",  "# = # * #", {1, 2, 3}},
-		{ "mulss",  "# = # * #", {1, 2, 3}},
-		{ "neg",  "# ~= #", {1, 1}},
-		{ "nop",  "", {0}},
-		{ "not",  "# = !#", {1, 1}},
-		{ "or",   "# |= #", {1, 2}},
-		{ "out",  "io[#] = #", {1, 2}},
-		{ "pop",  "pop #", {1}},
-		{ "push", "push #", {1}},
-		{ "ret",  "return", {0}},
-		{ "sal",  "# <<= #", {1, 2}},
-		{ "sar",  "# >>= #", {1, 2}},
-		{ "sete",  "# = e", {1}},
-		{ "setne",  "# = ne", {1}},
-		{ "shl",  "# <<<= #", {1, 2}},
-		{ "shld",  "# <<<= #", {1, 2}},
-		{ "sbb",  "# = # - #", {1, 1, 2}},
-		{ "shr",  "# >>>= #", {1, 2}},
-		{ "shlr",  "# >>>= #", {1, 2}},
-		//{ "strd",  "# = # - #", {1, 2, 3}},
-		{ "sub",  "# -= #", {1, 2}},
-		{ "swap", "var = #; # = #; # = var", {1, 1, 2, 2}},
-		{ "test", "var = # & #", {1, 2}},
-		{ "xchg",  "#,# = #,#", {1, 2, 2, 1}},
-		{ "xadd",  "#,# = #,#+#", {1, 2, 2, 1, 2}},
-		{ "xor",  "# ^= #", {1, 2}},
-		{ NULL }
-	};
 
 	if (argc > 2 && !strcmp (argv[0], "xor")) {
 		if (!strcmp (argv[1], argv[2])) {
 			argv[0] = "mov";
 			argv[2] = "0";
 		}
 	}
+#if USE_HT
+	static HtPP *ht = NULL;
```

Comment:
No static please.

Reply:
yeah, totally aware of this. the idea behind this pr was to open a discussion on how this optimization affects performance and where to cache/hold this info instead of creating it everytime we request it.